### PR TITLE
Unit test 2.0

### PR DESCRIPTION
**Pull request by timkimadobe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by cacheung** [comment]
Are you going to remove powermock for EdgeBridgeTest file?

File: code/edgebridge/build.gradle [hunk 1]
```diff
@@ -213,6 +213,7 @@ dependencies {
 
     testImplementation "androidx.test.ext:junit:${rootProject.ext.junitVersion}"
     testImplementation "org.mockito:mockito-core:${rootProject.ext.mockitoCoreVersion}"
+    testImplementation 'org.mockito:mockito-inline:4.5.1'
     testImplementation 'org.powermock:powermock-api-mockito2:2.0.0'
     testImplementation 'org.powermock:powermock-module-junit4:2.0.0'
```
Comment (emdobrin) [inline code/edgebridge/build.gradle:218]: can be removed, right?

File: code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java [hunk 2]
```diff
@@ -166,7 +158,14 @@ void handleRulesEngineResponse(@NonNull final Event event) {
 	private void dispatchTrackRequest(final Map<String, Object> data, final long timestamp) {
 		Map<String, Object> xdmData = new HashMap<>();
 		xdmData.put("eventType", EdgeBridgeConstants.JsonValues.EVENT_TYPE);
-		xdmData.put("timestamp", iso8601DateFormat.format(new Date(timestamp)));
+		xdmData.put(
+			"timestamp",
+			TimeUtils.getIso8601Date(
+				new Date(timestamp),
+				"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'",
+				TimeZone.getTimeZone("Etc/UTC")
```
Comment (emdobrin) [inline code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java:166]: we should keep the GMT timezone

Comment (emdobrin) [reply code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java:166]: also why not hiding the format and timezone under the TimeUtils API like we do in swift, ```e.g.: TimeUtils.getISO8601UTCDateWithMilliseconds(new Date(timeout))```

File: code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java [hunk 3]
```diff
@@ -42,67 +38,102 @@
...
-
+		Mockito.reset(mockExtensionApi);
 		extension = new EdgeBridgeExtension(mockExtensionApi);
 	}
 
 	// ========================================================================================
 	// constructor
 	// ========================================================================================
+	@Test
+	public void test_registerExtension_deprecated() {
```
Comment (emdobrin) [inline code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java:66]: - how are we testing the new method of registration?
- can we move these tests to EdgeBridgeTests since they test the testing the public APIs and not the internal EdgeBridgeExtension?

File: code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java [hunk 4]
```diff
@@ -136,6 +167,9 @@ public void test_getVersion() {
 	// ========================================================================================
 	// handleTrackRequest
 	// ========================================================================================
+	// remove mobileCore.dispatch mock in favor of ExtensionApi dispatch capture
+	// the "deprecated" way cannot be tested unless function/integration test because ExtensionApi itself is mocked
+	// and dispatching events goes through the ExtensionApi now instead of MobileCore
```
Comment (emdobrin) [inline code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java:172]: this seems to be a generic note, do you intend to keep it?

File: code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java [hunk 5]
```diff
@@ -42,67 +38,102 @@
...
-				callbackCaptor.capture()
-			);
-		assertNotNull("The extension callback for Rules Engine listener should not be null", callbackCaptor.getValue());
+		// Extract captured values into lists
+		List<String> eventTypes = eventTypeCaptor.getAllValues();
+		List<String> eventSources = eventSourceCaptor.getAllValues();
+		List<ExtensionEventListener> extensionEventListenerList = extensionEventListenerArgumentCaptor.getAllValues();
+
+		// Verify: 1st Edge Bridge event listener
+		assertEquals(EdgeBridgeTestConstants.EventType.GENERIC_TRACK, eventTypes.get(0));
```
Comment (emdobrin) [inline code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java:129]: for all these asserts can we use the core eventype/source instead?

Comment (emdobrin) [reply code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java:129]: I think for your tests you can assume core value will not change since they are publicly shared constants. In this case catching unexpectedly changed values should be part of core tests

File: code/edgebridge/build.gradle [hunk 6]
```diff
@@ -213,8 +213,7 @@ dependencies {
 
     testImplementation "androidx.test.ext:junit:${rootProject.ext.junitVersion}"
     testImplementation "org.mockito:mockito-core:${rootProject.ext.mockitoCoreVersion}"
-    testImplementation 'org.powermock:powermock-api-mockito2:2.0.0'
-    testImplementation 'org.powermock:powermock-module-junit4:2.0.0'
+    testImplementation 'org.mockito:mockito-inline:4.5.1'
```
Comment (emdobrin) [inline code/edgebridge/build.gradle:216]: do these libs evolve together, can we use rootProject.ext.mockitoVersion for both core and inline?

File: code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java [hunk 7]
```diff
@@ -253,8 +217,8 @@ public void test_handleTrackRequest_withEmptyEventData_doesNotDispatchEvent() {
 	public void test_handleRulesEngineResponse_dispatchesEdgeRequestEvent() {
 		final Event event = new Event.Builder(
 			"Test Rules Engine Event",
-			EdgeBridgeTestConstants.EventType.RULES_ENGINE,
-			EdgeBridgeTestConstants.EventSource.RESPONSE_CONTENT
```
Comment (emdobrin) [inline code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java:257]: can we also remove these values from EdgeBridgeTestConstants now?

File: code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java [hunk 8]
```diff
@@ -873,7 +759,7 @@ public void testGetExecutor_singleton() {
 	static {
 		final Locale posixLocale = new Locale(Locale.US.getLanguage(), Locale.US.getCountry(), "POSIX");
 		iso8601DateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", posixLocale);
-		iso8601DateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
+		iso8601DateFormat.setTimeZone(TimeZone.getTimeZone("Etc/UTC"));
```
Comment (emdobrin) [inline code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java:762]: this should be GMT

File: code/build.gradle [hunk 9]
```diff
@@ -20,7 +20,7 @@ ext {
 
     // dependencies
     junitVersion = "1.1.3"
-    mockitoCoreVersion = "2.23.0"
+    mockitoCoreVersion = "4.5.1"
```
Comment (emdobrin) [inline code/build.gradle:23]: nit: rename to mockitoVersion

**Review by emdobrin — COMMENTED** [review]
Thanks for the updates so far, changes look better. will be waiting for the pending comments before final review

File: code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java [hunk 10]
```diff
@@ -334,52 +298,42 @@ public void test_handleRulesEngineResponse_dispatchesEdgeRequestEvent() {
 		assertEquals(expectedData, responseEvent.getEventData());
 	}
 
-	@Test(expected = Test.None.class)
-	public void test_handleRulesEngineResponse_withNullEvent_doesNotThrow() {
```
Comment (cacheung) [inline code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java:338]: Don't need to test null this case?

File: code/edgebridge/src/androidTest/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeTests.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	@Test
+	public void test_registerExtension() throws InterruptedException {
+		final CountDownLatch latch = new CountDownLatch(1);
+		MobileCore.setLogLevel(LoggingMode.VERBOSE);
+		List<Class<? extends Extension>> extensions = new ArrayList<>();
+		extensions.add(EdgeBridge.EXTENSION);
+		MobileCore.registerExtensions(extensions, o -> latch.countDown());
+		assertTrue(latch.await(1000, TimeUnit.MILLISECONDS));
```
[the remark this answers is not in the document]
Comment (emdobrin) [reply code/edgebridge/src/androidTest/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeTests.java]: thanks for the updates, what you have right now under the unit tests looks good to me

**Review by emdobrin — APPROVED** [review]
